edit/buildozer.go

```diff
@@ -454,6 +455,85 @@ func cmdCopyNoOverwrite(opts *Options, env CmdEnvironment) (*build.File, error)
 	return copyAttributeBetweenRules(env, attrName, from)
 }
 
+// cmdDictAdd adds a key to a dict, if that key does _not_ exit already.
+func cmdDictAdd(opts *Options, env CmdEnvironment) (*build.File, error) {
+	attr := env.Args[0]
+	args := env.Args[1:]
+
+	dict := &build.DictExpr{}
+	currDict, ok := env.Rule.Attr(attr).(*build.DictExpr)
+	if ok {
+		dict = currDict
+	}
+
+	for _, x := range args {
+		kv := strings.Split(x, ":")
+		expr := build.StringExpr{
+			Value:       kv[1],
+			TripleQuote: false,
+			Token:       "",
+		}
+
+		prev := DictionaryGet(dict, kv[0])
+		if prev == nil {
+			// Only set the value if the value is currently unset.
+			DictionarySet(dict, kv[0], &expr)
+		}
+		env.Rule.SetAttr(attr, dict)
+	}
+	return env.File, nil
+}
+
+// cmdDictSet adds a key to a dict, overwriting any previous values.
+func cmdDictSet(opts *Options, env CmdEnvironment) (*build.File, error) {
+	attr := env.Args[0]
+	args := env.Args[1:]
+
+	dict := &build.DictExpr{}
+	currDict, ok := env.Rule.Attr(attr).(*build.DictExpr)
+	if ok {
+		dict = currDict
+	}
+
+	for _, x := range args {
+		kv := strings.Split(x, ":")
+		expr := build.StringExpr{
+			Value:       kv[1],
+			TripleQuote: false,
+			Token:       "",
+		}
+		// Set overwrites previous values.
+		DictionarySet(dict, kv[0], &expr)
+		env.Rule.SetAttr(attr, dict)
+	}
+	return env.File, nil
+}
+
+// cmdDictRemove removes a key from a dict.
+func cmdDictRemove(opts *Options, env CmdEnvironment) (*build.File, error) {
+	attr := env.Args[0]
+	args := env.Args[1:]
+
+	thing := env.Rule.Attr(attr)
+	dictAttr, ok := thing.(*build.DictExpr)
+	if ok {
+		for _, x := range args {
+			// should errors here be flagged?
+			DictionaryDelete(dictAttr, x)
+			env.Rule.SetAttr(attr, dictAttr)
+		}
+	}
+
+	// If the removal results in the dict having no contents, delete the attribute (stay clean!)
+	if dictAttr == nil || len(dictAttr.List) == 0 {
+		if env.Rule.DelAttr(attr) != nil {
+			return env.File, nil
```

Comment:
this seems like clutter
if dictAttr == nil || len(dictAttr.List) == 0 {
  env.Rule.DelAttr(attr)
}
return env.File, nil